### PR TITLE
Add providers to be removed from Bazel 8

### PR DESCRIPTION
**Pull request by comius**

### CODE REVIEW

**Comment by comius** [comment]
cc @hvadehra @rickeylev @ahumesky @timpeut @Wyverald @meteorcloudy

**Comment by meteorcloudy** [comment]
Can you link to a Bazel commit?

**Comment by comius** [comment]
> Can you link to a Bazel commit?

I can't because there isn't one yet.

This PR is needed to make changes in rules_java and rules_python. Once those are in, we can remove the providers from Bazel without breaking anything.

**Review by fmeum — APPROVED** [review]

**Comment by meteorcloudy** [comment]
I see, ship it!

File: private/globals.bzl [hunk 1]
```diff
@@ -20,4 +20,9 @@ GLOBALS = {
 # This one works in the reverse, put in the version when the global symbol is removed.
 LEGACY_GLOBALS = {
     "ProtoInfo": "8.0.0",
+    "JavaInfo": "8.0.0",
+    "JavaPluginInfo": "8.0.0",
+    "PyInfo": "8.0.0",
+    "PyRuntimeInfo": "8.0.0",
+    "PyCcLinkParamsProvider": "8.0.0",
```
Comment (brentleyjones) [inline private/globals.bzl:27]: We should keep this alphabetical.

Comment (fmeum) [reply private/globals.bzl:27]: Does `# keep sorted` work for dicts?

**Pull request merged**